main.cpp

```diff
@@ -77,7 +77,7 @@ int main()
     T t2(31, "Emma");                                             //6
     
     Compare f;                                            //7
-    auto smaller = f.compare(&t1, &t2);  //
+    auto smaller = f.compare(t1, t2);  //
```

Comment:
FIXME, `smaller` is still a pointer.  So you need to check if its safe to use before you use it!